From 6af8e1020f9dbfd7bdaca3b8b131acc7f5cf636e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 2 Nov 2024 11:21:52 +0100
Subject: [PATCH] Skip write-only formats

PALM and PDF are write-only formats by PIL, these cannot be opened.

Also move SVG out from the loop to avoid adding it multiple times.

Reference:
https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#write-only-formats
---
 cartridges/details_dialog.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/cartridges/details_dialog.py b/cartridges/details_dialog.py
index 8d85586a2..7ed67a4f0 100644
--- a/cartridges/details_dialog.py
+++ b/cartridges/details_dialog.py
@@ -90,8 +90,12 @@ def __init__(self, game: Optional[Game] = None, **kwargs: Any):
 
         image_filter = Gtk.FileFilter(name=_("Images"))
         for extension in Image.registered_extensions():
+            if extension[1:] in ["palm", "pdf"]:  # Skip write-only formats by PIL
+                continue
+
             image_filter.add_suffix(extension[1:])
-            image_filter.add_suffix("svg")  # Gdk.Texture supports .svg but PIL doesn't
+
+        image_filter.add_suffix("svg")  # Gdk.Texture supports .svg but PIL doesn't
 
         image_filters = Gio.ListStore.new(Gtk.FileFilter)
         image_filters.append(image_filter)
